fracas/map.cpp

```diff
@@ -823,9 +823,9 @@ void ARegionList::MakeUWMaze(ARegionArray *pArr)
             ARegion *reg = pArr->GetRegion(x, y);
             if (!reg) continue;
 
-            for (int i=D_NORTH; i<= NDIRS; i++) {
```

Comment:
Here, `i` will be NDIRS in the last iteration of the loop, and access reg->neighbors[NDIRS], which is one past the end of the array. By breaking the loop once i reaching NDIR, the bug is avoided.